### ISSUE TITLE
Data should not be compiled into models

### ISSUE THREAD

**Issue by Zaharid** [HEP-PBSP/colibri#9]
The current structure of the code does things along the lines of

```
def get_compiled_function():
    large_data = load_large_data()
    @jax.jit
    def func(input):
        return compute(large_data, input)
    return func

result = get_compiled_function()(my_input)

```

For example a relatively simple `weight_minimization_fit` exercise takes a large amount of time to build on my laptop (large fraction of one hour). 

The problem with that is that compiling the model, i.e. putting `large_data` inside of `func` is very expensive.

A more efficient system would be

```
def get_compiled_function():
    @jax.jit
    def func(input, large_data):
        return compute(large_data, input)
    return func

result = get_compiled_function()(my_input, large_data)
```

The reason this is not done is that coming up with a way to do batches with correlations, masks and so on, is a rather big challenge.
